src/Service/Models/MagicLinkTokenRequest.cs

```diff
@@ -0,0 +1,24 @@
+using System.Net.Mail;
+
+namespace Passwordless.Service.Models;
+
+/// <summary>
+/// Used to generate and send out the magic link email.
+/// </summary>
+/// <param name="userId">User identifier for the recipient of the magic link.</param>
+/// <param name="emailAddress">Email address for the intended user.</param>
+/// <param name="linkTemplate">Template used for creating the magic link. Token template string will be swapped with actual token.</param>
+/// <param name="timeToLive">Time span the magic link will be valid for. Default lifespan is 15 minutes.</param>
+public class MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, TimeSpan timeToLive) : RequestBase
+{
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate)
+        : this(userId, emailAddress, linkTemplate, new TimeSpan(0, 15, 0)) { }
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, int timeToLive)
+        : this(userId, emailAddress, linkTemplate, new TimeSpan(0, 0, timeToLive)) { }
+
+    public string UserId { get; } = userId;
+    public MailAddress EmailAddress { get; } = emailAddress;
+    public string LinkTemplate { get; } = linkTemplate;
+    public TimeSpan TimeToLive { get; } = timeToLive;
```

Comment:
To simplify, consider making `TimeSpan timeToLive` nullable.

Then update the property like so:

```csharp
public TimeSpan TimeToLive { get; } = timeToLive ?? TimeSpan.FromMinutes(15);
```

This will make the default more obvious, will simplify the constructors, and will remove the need for ternary in `SendMagicLinkRequest.ToDto()`.